src/analysis/retail/monk/mistweaver/modules/spells/ThunderFocusTea.tsx

```diff
@@ -72,31 +73,37 @@ class ThunderFocusTea extends Analyzer {
       this.tftCast,
     );
     this.addEventListener(Events.cast.by(SELECTED_PLAYER), this.buffedCast);
-    if (secretInfusionRank > 0) {
-      this.correctSpells = [
-        TALENTS_MONK.RENEWING_MIST_TALENT.id,
-        TALENTS_MONK.ESSENCE_FONT_TALENT.id,
-      ]; // only haste buff
-      if (this.ftActive) {
-        this.okSpells = [TALENTS_MONK.RISING_SUN_KICK_TALENT.id];
-      }
-    } else if (this.selectedCombatant.hasTalent(TALENTS_MONK.RISING_MIST_TALENT)) {
-      this.correctSpells = [
-        TALENTS_MONK.RISING_SUN_KICK_TALENT.id,
-        TALENTS_MONK.RENEWING_MIST_TALENT.id,
-      ];
-    } else {
-      this.correctSpells = [TALENTS_MONK.RENEWING_MIST_TALENT.id];
-      if (this.ftActive) {
-        this.okSpells = [TALENTS_MONK.RISING_SUN_KICK_TALENT.id];
-      }
-    }
+    this.okCapstoneSpells = this.selectedCombatant.hasTalent(TALENTS_MONK.RISING_MIST_TALENT)
+      ? [TALENTS_MONK.RISING_SUN_KICK_TALENT.id]
+      : [TALENTS_MONK.RENEWING_MIST_TALENT.id];
+    this.correctCapstoneSpells = this.selectedCombatant.hasTalent(TALENTS_MONK.RISING_MIST_TALENT)
```

Comment:
Not having rising mist is not sufficient enough for this imo, it needs to specifically check for ToM so that no weird fls builds etc slip through